lib/datastruct/patricia.c

```diff
@@ -144,7 +144,7 @@ compare(const struct pnode * n, const uint8_t * key, size_t keylen,
 	for (i = 0; i < n->slen; i++) {
 		/* Is the key a prefix of the node? */
 		if (keylen == i) {
-			*mlen = i;
+			*mlen = i & 0xff;
```

Comment:
in other circumstances, you've told me that `size_t i` should remain as such.  For example,

> So, the idea here is that a "straightforward" reading of the code should match what it actually does. i is an array index, so it should be a size_t; where we set a byte equal to i or 0x11 * i what we really mean is "set this byte to the lowest byte of..." and & 0xff is a way of making that explicit rather than simply relying on the implicit type conversion.
https://github.com/Tarsnap/libcperciva/pull/23#discussion_r68013973

I am *entirely* in favour of `uint8_t i`.  :)